Apps/WebClient/src/Server/Services/CommentService.cs

```diff
@@ -95,10 +95,47 @@ public RequestResult<IEnumerable<UserComment>> GetList(string hdId, string paren
                 };
             }
 
-            DBResult<IEnumerable<Comment>> dbComments = this.commentDelegate.GetList(hdId, parentEntryId);
+            DBResult<IEnumerable<Comment>> dbComments = this.commentDelegate.GetByParentEntry(hdId, parentEntryId);
             RequestResult<IEnumerable<UserComment>> result = new RequestResult<IEnumerable<UserComment>>()
             {
                 ResourcePayload = UserComment.CreateListFromDbModel(dbComments.Payload, this.cryptoDelegate, key),
+                TotalResultCount = dbComments.Payload.Count(),
+                PageIndex = 0,
+                PageSize = dbComments.Payload.Count(),
+                ResultStatus = dbComments.Status == DBStatusCode.Read ? ResultType.Success : ResultType.Error,
+                ResultError = dbComments.Status != DBStatusCode.Read ? new RequestResultError() { ResultMessage = dbComments.Message, ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationInternal, ServiceType.Database) } : null,
+            };
+            return result;
+        }
+
+        /// <inheritdoc />
+        public RequestResult<IDictionary<string, IEnumerable<UserComment>>> GetProfileComments(string hdId)
```

Comment:
we may need unit test for this new method as well.

Reply:
same as above